Framework/LiveData/src/LoadLiveData.cpp

```diff
@@ -500,6 +500,23 @@ void LoadLiveData::exec() {
     // Default to Add.
     this->addChunk(processed);
 
+  // If m_accumWS is an EventWorkspace, we have to update the bin boundaries
+  if (EventWorkspace_sptr accumEvent =
+          boost::dynamic_pointer_cast<EventWorkspace>(m_accumWS)) {
+    // Calculate fresh min and max based on all events in WS
+    double tofmin, tofmax;
+    accumEvent->getEventXMinMax(tofmin, tofmax);
+
+    // Ugly hack to update all X values in a performant way.
```

Comment:
You should reference the issue we discussed that you should create for the bin edges problem in the comment too. You have profiling information that we don't want to loose, and want to connect up with this hack.